Paintroid/src/androidTest/java/org/catrobat/paintroid/test/espresso/tools/ShapeToolIntegrationTest.java

```diff
@@ -120,4 +120,51 @@ public void testRememberShapeAfterOrientationChange() {
 
 		assertTrue(expectedBitmap.sameAs(PaintroidApplication.drawingSurface.getBitmapCopy()));
 	}
+
+	@Test
+	public void testRememberOutlinePositionAfterOrientationChange() {
+		onView(withId(R.id.shape_ibtn_outline))
+				.perform(click());
+		onView(withId(shape))
+				.perform(click());
+
+		PointF p = new PointF(0, 0);
+		((BaseToolWithShape) PaintroidApplication.currentTool).toolPosition.set(p);
+
+		activityTestRule.getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
+
+		assertEquals(p, ((BaseToolWithShape) PaintroidApplication.currentTool).toolPosition);
+	}
+
+	@Test
+	public void testRememberOutlineShapeAfterOrientationChange() {
+		onView(withId(R.id.shape_ibtn_outline))
+				.perform(click());
+		onView(withId(shape))
+				.perform(click())
+				.check(matches(isSelected()));
+		onToolBarView()
+				.performCloseToolOptions();
+
+		onView(isRoot())
+				.perform(click());
```

Comment:
This might click on the wrong position. I would prefer to use 
```
onDrawingSurfaceView()
		.perform(touchAt(DrawingSurfaceLocationProvider.TOOL_POSITION));
```

This should then stamp correctly.